pkg/operation/botanist/component/kubeapiserver/secrets.go

```diff
@@ -235,17 +235,21 @@ func aesKeyFromSecretData(data map[string][]byte) apiserverconfigv1.Key {
 func (k *kubeAPIServer) reconcileSecretServer(ctx context.Context) (*corev1.Secret, error) {
 	var (
 		ipAddresses = append([]net.IP{}, k.values.ServerCertificate.ExtraIPAddresses...)
-		dnsNames    = append([]string{
+		dnsNames    = []string{
 			v1beta1constants.DeploymentNameKubeAPIServer,
 			fmt.Sprintf("%s.%s", v1beta1constants.DeploymentNameKubeAPIServer, k.namespace),
 			fmt.Sprintf("%s.%s.svc", v1beta1constants.DeploymentNameKubeAPIServer, k.namespace),
-		}, kutil.DNSNamesForService("kubernetes", metav1.NamespaceDefault)...)
+		}
 	)
 
 	if k.values.SNI.PodMutatorEnabled || k.values.VPN.HighAvailabilityEnabled {
 		ipAddresses = append(ipAddresses, net.ParseIP("127.0.0.1"))
 	}
 
+	if k.values.IsNodeless {
```

Comment:
```suggestion
        if !k.values.IsNodeless {
```
If I'm not completely mistaken 😄

Reply:
Nice finding!